src/OpenTelemetry/Trace/SimpleExportActivityProcessor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// <copyright file="SimpleExportActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Implements simple activity processor that exports activities in OnEnd call without batching.
+    /// </summary>
+    public class SimpleExportActivityProcessor : ActivityProcessor
+    {
+        private readonly ActivityExporterSync exporter;
+        private bool stopped;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleExportActivityProcessor"/> class.
+        /// </summary>
+        /// <param name="exporter">Activity exporter instance.</param>
+        public SimpleExportActivityProcessor(ActivityExporterSync exporter)
+        {
+            this.exporter = exporter ?? throw new ArgumentNullException(nameof(exporter));
+        }
+
+        /// <inheritdoc />
+        public override void OnEnd(Activity activity)
+        {
+            try
+            {
+                // TODO: avoid heap allocation
+                _ = this.exporter.Export(new[] { activity });
+            }
+            catch (Exception ex)
+            {
+                OpenTelemetrySdkEventSource.Log.SpanProcessorException(nameof(this.OnEnd), ex);
+            }
+        }
+
+        /// <inheritdoc />
+        public override Task ShutdownAsync(CancellationToken cancellationToken)
+        {
+            if (!this.stopped)
+            {
+                this.exporter.Shutdown();
```

Comment:
No need for try/catch here?

Reply:
@reyang I merged to unblock you, but probably need try/catch in a future PR?

Reply:
I need to get feedback here.

The spec is saying that [exporter should not throw](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#basic-error-handling-principles).

I had similar struggle [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/a4237cf104482332bea006acf0c6610dfee71e61/src/OpenTelemetry/Trace/CompositeActivityProcessor.cs#L78), whether should I trust the other processors won't throw, or I should make the composite one bullet proof.

Reply:
So the higher order question is, how much do we trust / rely on the other components (whether processors/exporters that we own / in this repo, or we don't own / in other repos) to follow the spec / be a good citizen.

Reply:
Here is my take while I was a C/C++ runtime library dev:

1. If a developer calls `free` on a buffer (allocated by malloc) that is already freed, we want to fail him instead of keep silent. In this way we are actually helping that developer to discover his bug, instead of hiding it as the bug could potentially has other implications (e.g. data loss, corruption, logical flaw).
2. If the heap manager observed any sign of an incorrect heap (e.g. via integrity check on sentry bytes), it should crash the app ASAP instead of trying to hide the mistake (since the app might run on dangling pointers which writes dirty data to a database/file).

However, here I'm struggling as I believe we have another design principle - a telemetry SDK, once initialized, should not punish the users by killing their applications.

Reply:
Even good citizens screw up. I know I have (though unsure if I'm _good_).

Is there be a perf or some other concern to being defensive at these integration points?

Reply:
It is mainly from the responsibility and dev experience perspective.

For example if some bad guy developed a broken exporter that throws (against the spec law) and we swallowed the exception for him, it seems like we are contributing to a crime instead of reporting to the cops.

And if a good guy is developing an exporter where he made subtle mistake, by throwing error directly we helped him to fix his bug. Otherwise he will have to look at the self diagnostic logs.

Anyways, I'm struggling here which means I don't have strong preference (and as you can see, I do have strong preference that we should catch Dispose exception and I put that catch statement 😄). So I'm happy to change either way (or even have a global flag to turn on/off such behavior) if @alanwest @CodeBlanch or anyone in the community has a strong preference.

Reply:
As usual, @reyang raises good points! I'm good with letting it throw. But I will say this... that doesn't necessarily guarantee anyone will pay attention! I have seen a lot of code get to prod where ops immediately starts complaining things won't shut down. Launch the app in dev, it is easily reproducable. How does this get by the devs? Because they use the "stop" button in VS which doesn't perform graceful shutdown 🙄

Reply:
Yes, definitely good points to ponder. I'm also good with letting it throw.

Though, I've generally thought of the telemetry SDK and other profiling tools as similar to logging frameworks where "do no harm" is often the general mantra. Usually (hopefully) there are other signs that something is awry - like a sudden drop of data reported to your backend, even though seemingly your app is still running "just fine".

This conversation is definitely bigger than just exporter or processors - these questions would apply to instrumentation among other things as well. It's more of general design philosophy, maybe more appropriate for a spec-level kind of guidance/adoption?

Reply:
A slightly different story from the [Nt vs. Zw API](https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/using-nt-and-zw-versions-of-the-native-system-services-routines).

I think given that none of us have strong opinion, and it seems @alanwest @CodeBlanch both have/had the intuition that it is nicer to swallow exception, we probably should go with the direction (to swallow exceptions) for now and make sure we handle this consistently across the repo?

Reply:
If I got two 👍 I'll send a PR to swallow the exception for both the exporting processor and composite processor.